GlobalQuakeClient/src/main/java/globalquake/ui/globalquake/EarthquakeListPanel.java

```diff
@@ -77,15 +87,34 @@ public void mousePressed(MouseEvent e) {
 
                 ArchivedQuake quake = filtered.get(i);
 
+                UUID quakeUuid = quake.getUuid();
+
                 if (quake != null && e.getButton() == MouseEvent.BUTTON3 && !isMouseInGoUpRect) {
                     quake.setWrong(!quake.isWrong());
                 }
 
                 if(e.getButton() == MouseEvent.BUTTON1) {
                     if(isMouseInGoUpRect) {
                         scroll = 0;
-                    }else if (quake != null ) {
-                        new ArchivedQuakeUI(parent, quake).setVisible(true);
+                    }
+                    else if (quake != null && !isOpened(quakeUuid) ) {
+                        // Create Instance of quakeUi panel
+                        ArchivedQuakeUI quakeUi = new ArchivedQuakeUI(parent, quake);
+                        // set the panel to be visible
+                        quakeUi.setVisible(true);
+                        // add the quake to the list of opened panels
+                        openedQuakes.add(quakeUuid);
+
+                        // When quake panel is closed, remove quake from the ArrayList of openedQuakes
+                        quakeUi.addWindowListener(new WindowAdapter() {
+                            @Override
+                            public void windowClosing(WindowEvent e) {
+                                openedQuakes.remove(quakeUuid);
+                            }
+                            public void windowClosed(WindowEvent e) {
+                                openedQuakes.remove(quakeUuid);
```

Comment:
Do we need to remove the uuid twice on closing and closed event? 
is there a case where 1 of these events doesnt do what it is supposed to?

Reply:
This is pretty much for redundancy, I did research and I found it is best to include both!

Reply:
All good, just wanted to know the reason around it.